Updated checkpoint handling to enable zWGRv1 spending

Kokary opened the pull request.

Kokary committed:
Updated checkpoint handling to enable zWGRv1 spending

ckti reviewed (commented):
Looks like the code is a bit less cluttered as well

Cryptarchist commented:
Code looks good, I will build and test the zWGRv1 spending stuff.

Nit outside the scope of this PR is whether the command should be named `calculateaccumulatorvalues` (<-- command name) or `generateaccumulatorvalues` (<-- help text name).

Kokary commented:
No you're right, I updated the help wording.

(FYI, it's a helper function that I added to generate the json.h file, and introduces an alternative to the existing getaccumulatorvalues function: one that doesn't use existing checkpoints or the local database. Wasn't sure if we should keep it, but I figured: why not?)

Cryptarchist commented:
@Kokary I think keep it. I played around with it earlier when I was testing this PR, took a few CPU cycles for it to calculate the values (thought it was broken), but it eventually spat out the matching results to match those in `accumulatorcheckpoints.json.h`.

Cryptarchist commented on src/accumulatorcheckpoints.json.h:
```diff
@@ -10,26 +10,345 @@
-            "    \"500\":  \"3c1\",\n"
-            "    \"1000\": \"3c1\",\n"
-            "    \"5000\": \"3c1\"\n"
+            "    \"height\": 10020,\n"
```
The values for height `10020` seem to be duplicated on line 35.

Cryptarchist commented on src/accumulatorcheckpoints.json.h:
```diff
@@ -10,26 +10,345 @@
+            "    \"5000\": \"154170d673be7e543e84107cf23d6a7d114a6f2b47701c414acb2a9c0c42e138a1e2d74cb2fe80b6c6a760a4025f2e024e3785cad1b0bc3346967d59c340fabe99406bb4adfc1b8d4423511641fe532ac685697e19ab9a96919f92470ba956a9f8d285185ec0f198f6700374d5e8ba69438b098849e85c374f744221dbe3a5360e6fef7bc2c737a1aacf65afb49e078288079d334dc71aae73e18059124587344dfeecde259d47aa0d075eb95ac3688bed1e6eb445bf8a9042 ...
+            "  },\n"
+            "  {\n"
+            "    \"height\": 20020,\n"
```
The values for height `20020` seem to be duplicated on line 46.

Kokary commented:
Thanks, removed duplicate values.

Cryptarchist commented:
zWGRv1 spending confirmed working. ACK 58be8893fafa9005d9bf151df792af111368e2e0
